src/SubPoolFactory.sol

```diff
@@ -0,0 +1,193 @@
+        if (subpoolData.exitTimestamp == 0) revert SubPoolFactory__PoolHasNotAnnouncedExitYet();
+        if (newExitTimestamp > subpoolData.exitTimestamp) revert SubPoolFactory__InvalidFastTrackExit();
+        subpoolData.exitTimestamp = newExitTimestamp;
+        subPoolData[pool] = subpoolData;
+    }
+
+    /// @notice Update solver membership.
+    function updateSolverMembership(address solver, bool add) external {
+        address pool = msg.sender;
+        SubPoolData memory subpoolData = subPoolData[pool];
+        if (subpoolData.collateral == address(0)) revert SubPoolFactory__UnknownPool();
+
+        if (add) {
+            // check if solver has a membership of some other pool
+            if (solverBelongsTo[solver] != address(0)) revert SubPoolFactory__SolverHasActiveMembership();
+            solverBelongsTo[solver] = pool;
+            emit UpdateSolverMembership(pool, solver, true);
+        } else {
+            // check that the solver is member of the given pool
+            if (solverBelongsTo[solver] != pool) revert SubPoolFactory__SolverNotAMember();
+            solverBelongsTo[solver] = address(0);
+            emit UpdateSolverMembership(pool, solver, false);
+        }
+    }
+
+    /// @notice Leave a subpool membership.
+    function leaveSubPool() external {
+        address pool = solverBelongsTo[msg.sender];
+        solverBelongsTo[msg.sender] = address(0);
+        emit UpdateSolverMembership(pool, msg.sender, false);
+    }
+
+    /// @notice Determine if the solver can submit solutions.
+    function canSolve(address solver) external view returns (bool) {
+        address pool = solverBelongsTo[solver];
+
+        SubPoolData memory subpoolData = subPoolData[pool];
+        // check first if the pool exists
+        if (subpoolData.collateral == address(0)) revert SubPoolFactory__UnknownPool();
+
+        // cannot solve if announced exit
+        if (subpoolData.exitTimestamp != 0) {
+            return false;
+        }
+
+        // cannot solve if there are any pending dues
+        (uint256 amt, uint256 cowAmt, uint256 ethAmt) = SubPool(pool).dues();
+        return amt == 0 && cowAmt == 0 && ethAmt == 0;
+    }
+
+    /// @notice Read subpool's exit timestamp.
+    function exitTimestamp(address pool) external view returns (uint256) {
+        SubPoolData memory subpoolData = subPoolData[pool];
+        if (subpoolData.collateral == address(0)) revert SubPoolFactory__UnknownPool();
+        return subpoolData.exitTimestamp;
+    }
+
+    /// @notice Read solver's subpool address.
+    /// @dev    Only difference from `poolOf` is this checks if the subpool has been initialized.
+    function solverSubPool(address solver) public view returns (address) {
+        address pool = poolOf(solver);
+        SubPoolData memory subpoolData = subPoolData[pool];
+        if (subpoolData.collateral == address(0)) revert SubPoolFactory__UnknownPool();
+        return pool;
+    }
```

Comment:
The signature of this function is a misnomer: the subpool of a solver is retrieved with `solverBelongsTo`.

```suggestion
    /// @notice Read solver's subpool address.
    /// @dev    Only difference from `poolOf` is this checks if the subpool has been initialized.
    function subPoolByCreator(address creator) public view returns (address) {
        address pool = poolOf(creator);
        SubPoolData memory subpoolData = subPoolData[pool];
        if (subpoolData.collateral == address(0)) revert SubPoolFactory__UnknownPool();
        return pool;
    }
```